.pre-commit-config.yaml

```diff
@@ -68,13 +63,13 @@ repos:
     hooks:
       - id: basedpyright
         name: basedpyright
-        entry: poetry run basedpyright
+        entry: uv run basedpyright
         language: system
         types_or: [python, pyi]
 
       - id: basedpyright-verifytypes
         name: basedpyright --verifytypes
-        entry: poetry run basedpyright --ignoreexternal --verifytypes mainpy
+        entry: uv run basedpyright --ignoreexternal --verifytypes mainpy
         language: system
         always_run: true
         pass_filenames: false
```

Comment:
it would be nice to have a deploy job, but maybe that for another PR

Reply:
Yes, that's something that's missing from almost all of my repo's actually. So you think that building + publishing based on a tag-push is something that'll work?

Reply:
can do a strategy like https://github.com/KotlinIsland/basedmypy/issues/167

basically:
- create draft release on every merge to `main`
- on publish of draft release, trigger publish job

Reply:
needs https://github.com/astral-sh/uv/issues/6278

Reply:
there's an issue for this now: https://github.com/jorenham/mainpy/issues/45